src/core/import/qImport.ts

```diff
@@ -117,33 +95,40 @@ export function toImportPath(url: URL): string {
 /**
  * Convert custom protocol to path by looking it up in `QConfig`
  *
- * Pats such as
+ * Paths such as
  * ```
  * QRL`foo:/bar`
- *
- * Q = {
- *   protocol: {
- *     'foo': 'somePath'
- *   }
- * }
  * ```
- * The `QRL` looks up `foo` in `QRLProtocolMap` resulting in `somePath/bar`
  *
- * @param qConfig
+ * The `QRL` looks up `foo` in the document's `<link ref="q.protocol.foo" href="somePath">`
+ * resulting in `somePath/bar`
+ *
+ * @param doc
  * @param qrl
  * @returns URL where the custom protocol has been resolved.
  */
-export function adjustProtocol(qConfig: QConfig, qrl: string | QRL): string {
+function adjustProtocol(doc: Document, qrl: string | QRL): string {
   return String(qrl).replace(/(^\w+):\/?/, (all, protocol) => {
-    let value = qConfig.protocol[protocol];
-    if (!value) return all;
-    if (!value.endsWith('/')) {
+    let value = getConfig(doc, `protocol.` + protocol);
+    if (value && !value.endsWith('/')) {
       value = value + '/';
     }
-    return value;
+    return value || all;
   });
 }
 
+const configCache = new Map<string, string | undefined | null>();
```

Comment:
As it stands now, I think this will not behave correctly. On a server, one can serve different apps. These apps will have different configurations but may have the same protocol. The above will leak data between the applications. Let's remove this cache altogether.